tomcat/src/main/java/org/apache/coyote/http11/response/HttpResponse.java

```diff
@@ -2,37 +2,121 @@
         this.status = status;
+        this.location = location;
+        this.cookie = cookie;
         this.contentType = contentType;
         this.responseBody = responseBody;
     }
 
+    public static HttpResponseBuilder ok() {
+        return new HttpResponseBuilder()
+                .protocol(DEFAULT_PROTOCOL)
+                .status(HttpStatus.OK);
+    }
+
+    public static HttpResponseBuilder redirect() {
+        return new HttpResponseBuilder()
+                .protocol(DEFAULT_PROTOCOL)
+                .status(HttpStatus.FOUND);
+    }
+
+    public static HttpResponseBuilder notFound() {
+        return new HttpResponseBuilder()
+                .protocol(DEFAULT_PROTOCOL)
+                .status(HttpStatus.NOT_FOUND);
+    }
+
+    public static class HttpResponseBuilder {
+
+        private String protocol;
+        private HttpStatus status;
+        private String location;
+        private Cookie cookie;
+        private ContentType contentType;
+        private String responseBody;
+
+        public HttpResponseBuilder protocol(String protocol) {
+            this.protocol = protocol;
+            return this;
+        }
+
+        public HttpResponseBuilder status(HttpStatus status) {
+            this.status = status;
+            return this;
+        }
+
+        public HttpResponseBuilder addLocation(String location) {
+            this.location = location;
+            return this;
+        }
+
+        public HttpResponseBuilder addResponseBody(String responseBody, ContentType contentType) {
+            this.responseBody = responseBody;
+            this.contentType = contentType;
+            return this;
+        }
+
+        public HttpResponseBuilder addCookie(Cookie cookie) {
+            this.cookie = cookie;
+            return this;
+        }
+
+        public HttpResponse build() {
+            return new HttpResponse(protocol, status, location, cookie, contentType, responseBody);
+        }
+    }
+
     public String parseToString() {
-        return String.join("\r\n",
-                protocol + " " + status.getCode() + " ",
-                "Content-Type: " + contentType.getType() + " ",
-                "Content-Length: " + responseBody.getBytes().length + " ",
-                "",
-                responseBody);
+        StringBuilder sb = new StringBuilder();
+        if (protocol != null) {
+            sb.append(protocol).append(" ");
+        }
+        if (status != null) {
+            sb.append(status.getCode()).append(" ").append(status.getText()).append(" ");
+        }
+        sb.append("\r\n");
+
+        if (location != null) {
+            sb.append("Location: ").append(location).append("\r\n");
+        }
+        if (cookie != null) {
+            sb.append("Set-Cookie: ").append(cookie.parseToString()).append("\r\n");
+        }
+        if (responseBody != null) {
+            sb.append("Content-Type: ").append(contentType.getType()).append(" ").append("\r\n");
```

Comment:
```suggestion
            sb.append("Content-Type: ").append(contentType.getType()).append(" \r\n");
```
소소하지만 이렇게 변경되어도 될 것 같네요